<a href="https://colab.research.google.com/github/Vronska-Anhelina/-Prediction-and-Clustering-Models/blob/main/Credit_Scoring_with_Categorical_Variables.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
data = pd.DataFrame({
    "Client_ID": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
    "Employment_Type": ["Full-time", "Part-time", "Unemployed", "Full-time", "Part-time",
                        "Unemployed", "Full-time", "Part-time", "Unemployed", "Full-time"],
    "Marital_Status": ["Single", "Married", "Divorced", "Single", "Married",
                       "Single", "Divorced", "Married", "Single", "Divorced"],
    "Credit_Score": [750, 680, 600, 720, 710, 580, 770, 690, 620, 740]
})
print(data)

   Client_ID Employment_Type Marital_Status  Credit_Score
0          1       Full-time         Single           750
1          2       Part-time        Married           680
2          3      Unemployed       Divorced           600
3          4       Full-time         Single           720
4          5       Part-time        Married           710
5          6      Unemployed         Single           580
6          7       Full-time       Divorced           770
7          8       Part-time        Married           690
8          9      Unemployed         Single           620
9         10       Full-time       Divorced           740


In [ ]:
df=pd.get_dummies(data,columns=["Employment_Type","Marital_Status"],drop_first=True)
print(df)

   Client_ID  Credit_Score  Employment_Type_Part-time  \
0          1           750                      False   
1          2           680                       True   
2          3           600                      False   
3          4           720                      False   
4          5           710                       True   
5          6           580                      False   
6          7           770                      False   
7          8           690                       True   
8          9           620                      False   
9         10           740                      False   

   Employment_Type_Unemployed  Marital_Status_Married  Marital_Status_Single  
0                       False                   False                   True  
1                       False                    True                  False  
2                        True                   False                  False  
3                       False                   False   

In [ ]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression

X=df.drop(["Client_ID", "Credit_Score"],axis=1)
y=df["Credit_Score"]
X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.3,random_state=2026)
model=LogisticRegression()
model.fit(X_train,y_train)
y_pred=model.predict(X_test)
coefficients=pd.DataFrame({"Feature":X.columns,"Coefficients":model.coef_[0]}).sort_values(by="Coefficients",ascending=False)
print("Feature Importance:")
print(coefficients)

Feature Importance:
                      Feature  Coefficients
1  Employment_Type_Unemployed      0.401351
3       Marital_Status_Single      0.401351
0   Employment_Type_Part-time     -0.141276
2      Marital_Status_Married     -0.141276


We see that two of the four coefficients are negative. This indicates a negative impact of the categories "Part-time employment" and "Married" on the target variable compared to the base category Credit Score. Employment_Type_Unemployed (0.401) and Marital_Status_Single (0.401) increase the probability of a certain Credit Score class.